doc/source/python-tiledbsoma-io.rst

```diff
@@ -0,0 +1,22 @@
+The tiledbsoma.io module
+=====================
+
+.. currentmodule:: tiledbsoma.io
+
+.. automodule:: tiledbsoma.io
+
+
+Functions
+-------
+
+.. autosummary::
+    :toctree: _autosummary/
+    :nosignatures:
+
+    tiledbsoma.io.add_X_layer
+    tiledbsoma.io.add_matrix_to_collection
+    tiledbsoma.io.create_from_matrix
+    tiledbsoma.io.from_anndata
+    tiledbsoma.io.from_h5ad
+    tiledbsoma.io.to_anndata
+    tiledbsoma.io.to_h5ad
```

Comment:
```suggestion
    tiledbsoma.io.from_h5ad
    tiledbsoma.io.from_anndata
    tiledbsoma.io.add_X_layer
    tiledbsoma.io.add_matrix_to_collection
    tiledbsoma.io.create_from_matrix
    tiledbsoma.io.to_h5ad
    tiledbsoma.io.to_anndata
```

Idea:

`from_h5ad`, `from_anndata`, `to_h5ad`, and `to_anndata` are the "big four"

the rest are helpers for doing things like adding an `obsm` element